Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,321 @@
+            'EntryContext': {}
+        }
+
+        # Ask TOS for the path
+        o = tos_request("/securetrack/api/topology/path", querystring)
+
+        # Verify the data and return
+        try:
+            entry['EntryContext']['Tufin.TrafficAllowed'] = o['path_calc_results']['traffic_allowed']
+            entry['EntryContext']['Tufin.TrafficRawJSON'] = o['path_calc_results']['device_info']
+            entry['EntryContext']['Tufin.TrafficDevices'] = [d['name'] for d in o['path_calc_results']['device_info']]
+            entry['Contents'] = o['path_calc_results']['device_info']
+            entry['HumanReadable'] = tableToMarkdown('Tufin Topology Search for {} to {} via Service {}. Traffic is {}'.format(querystring['src'], querystring['dst'], querystring['service'], ('**Denied**', '**Allowed**')[o['path_calc_results']['traffic_allowed']]),
+            {'Start': querystring['src'], 'Devices in Path' : "-->".join(['**' + d['name'] + '**' + ' ({})'.format(d['vendor']) for d in o['path_calc_results']['device_info']]), 'End': querystring['dst']}, ['Start', 'Devices in Path', 'End'])
+        except KeyError:
+            entry['HumanReadable'] = 'Unknown Output Returned'
+            entry['Contents'] = 'Unknown Output Returned'
+            entry['ContentsFormat'] = formats['text']
+            entry['Type'] = entryTypes['error']
+            return entry
+
+        return entry
+
+
+    def path_finder_image():
+        """ Sample query: querystring = {"src":"10.80.80.0","dst":"172.16.200.80","service":"tcp:80","includeIncompletePaths":"true","displayBlockedStatus":"true"} """
+        querystring = {
+            "src": demisto.args()['Source'],
+            "dst": demisto.args()['Destination'],
+            "service": demisto.args().get('Service', 'Any'),
+            "includeIncompletePaths": "true",
+            "displayBlockedStatus": "true"
+        }
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['text'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        try:
+            img = tos_request("/securetrack/api/topology/path_image", querystring, {'accept': "image/png",'content-type': "application/json",'cache-control': "no-cache"})
+            if len(img) > 100:
+                return fileResult('topo.png', img, entryTypes['image'])
+            else:
+                entry['HumanReadable'] = 'No Valid Path Found'
+                entry['Contents'] = 'No Valid Path Found'
+                return entry
+        except:
+            entry['HumanReadable'] = 'Error Running Query'
+            entry['Contents'] = 'Error Running Query'
+            entry['Type'] = entryTypes['error']
+            return entry
+
+
+    def device_name(devices, device_id):
+        return [e['name'] + ' ({} {})'.format(e['vendor'], e['model']) for e in devices if int(e['id']) == int(device_id)][0]
```

Comment:
Please split to a few lines to make code a bit more readable.

Reply:
Breaking up the list comprehension may not be any more readable.